src/control/server/harness_test.go

```diff
@@ -357,3 +363,128 @@ func TestHarnessIOServerStart(t *testing.T) {
 		})
 	}
 }
+
+func TestHarness_StopInstances(t *testing.T) {
+	for name, tc := range map[string]struct {
+		ioserverCount     int
+		missingSB         bool
+		signal            os.Signal
+		ranks             []*ioserver.Rank
+		harnessNotStarted bool
+		signalErr         error
+		ctxTimeout        time.Duration
+		expResults        system.MemberResults
+		expSignalsSent    map[uint32]os.Signal
+		expErr            error
+	}{
+		"nil signal": {
+			expErr: errors.New("nil signal"),
+		},
+		"missing superblock": {
+			missingSB: true,
+			signal:    syscall.SIGKILL,
+			expErr:    errors.New("nil superblock"),
+		},
+		"harness not started": {
+			harnessNotStarted: true,
+			signal:            syscall.SIGKILL,
+			expSignalsSent:    map[uint32]os.Signal{},
+		},
+		"rank not in list": {
+			ranks:          []*ioserver.Rank{ioserver.NewRankPtr(2), ioserver.NewRankPtr(3)},
+			signal:         syscall.SIGKILL,
+			expSignalsSent: map[uint32]os.Signal{1: syscall.SIGKILL}, // instance 1 has rank 2
+		},
+		"signal send error": {
+			signal:    syscall.SIGKILL,
+			signalErr: errors.New("sending signal failed"),
+			expErr:    errors.New("scan error(s):\n  sending signal failed\n  sending signal failed\n"),
+		},
+		"context timeout": {
+			signal:     syscall.SIGKILL,
+			ctxTimeout: 1 * time.Nanosecond,
+			expErr:     errors.New("context deadline exceeded"),
+		},
+		"process wait takes too long": {
+			signal:     syscall.SIGKILL,
+			ctxTimeout: testShortTimeout,
+			expErr:     errors.New("context deadline exceeded"),
+		},
```

Comment:
Aren't these two the same?

Reply:
This one doesn't seem to be resolved with the latest revision. Is there something between these two cases that differs? Both exceed the context deadline, by passing a short timeout. Am I missing something?